src/main/resources/com/puppycrawl/tools/checkstyle/grammars/javadoc/JavadocLexer.g4

```diff
@@ -120,10 +120,9 @@ Newline5: NEWLINE
 Leading_asterisk3: LEADING_ASTERISK -> type(LEADING_ASTERISK);
 XmlTagOpen1: '<' -> type(OPEN), pushMode(xmlTagDefinition);
 STRING: '"' .*? '"' {referenceCatched = false;} -> mode(DEFAULT_MODE);
-PACKAGE: [a-z_$] ([a-z0-9_$] | '.')+ [a-z0-9_$] {referenceCatched = true;};
-DOT: '.';
+PACKAGE_CLASS: Identifier ('.' Identifier)* {_input.LA(1) == '#' || _input.LA(1) == '}'
```

Comment:
The last tag in the file I mentioned would be it. This one: `{@value lang.Integer#MAX_VALUE}`